src/main/java/jcifs/internal/smb1/trans/nt/NtTransQuerySecurityDescResponse.java

```diff
@@ -0,0 +1,109 @@
+/* jcifs smb client library in Java
+ * Copyright (C) 2005  "Michael B. Allen" <jcifs at samba dot org>
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package jcifs.internal.smb1.trans.nt;
+
+
+import java.io.IOException;
+
+import jcifs.Configuration;
+import jcifs.RuntimeCIFSException;
+import jcifs.internal.dtyp.SecurityDescriptor;
+import jcifs.internal.util.SMBUtil;
+
+
+/**
+ * 
+ */
+public class NtTransQuerySecurityDescResponse extends SmbComNtTransactionResponse {
+
+    private SecurityDescriptor securityDescriptor;
+
+
+    /**
+     * 
+     * @param config
+     */
+    public NtTransQuerySecurityDescResponse ( Configuration config ) {
+        super(config);
+    }
+
+
+    /**
+     * @return the securityDescriptor
+     */
+    public final SecurityDescriptor getSecurityDescriptor () {
+        return this.securityDescriptor;
+    }
+
+
+    @Override
+    protected int writeSetupWireFormat ( byte[] dst, int dstIndex ) {
+        return 0;
+    }
+
+
+    @Override
+    protected int writeParametersWireFormat ( byte[] dst, int dstIndex ) {
+        return 0;
+    }
+
+
+    @Override
+    protected int writeDataWireFormat ( byte[] dst, int dstIndex ) {
+        return 0;
+    }
+
+
+    @Override
+    protected int readSetupWireFormat ( byte[] buffer, int bufferIndex, int len ) {
+        return 0;
+    }
+
+
+    @Override
+    protected int readParametersWireFormat ( byte[] buffer, int bufferIndex, int len ) {
+        this.length = SMBUtil.readInt4(buffer, bufferIndex);
+        return 4;
+    }
+
+
+    @Override
+    protected int readDataWireFormat ( byte[] buffer, int bufferIndex, int len ) {
+        int start = bufferIndex;
+
+        if ( this.getErrorCode() != 0 )
+            return 4;
+
+        try {
+            this.securityDescriptor = new SecurityDescriptor();
+            bufferIndex += this.securityDescriptor.decode(buffer, bufferIndex, len);
+        }
+        catch ( IOException ioe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)